frontend/scripts/actions/app.actions.js

```diff
@@ -17,6 +22,52 @@ export const TOGGLE_MAP_LAYERS_MENU = 'TOGGLE_MAP_LAYERS_MENU';
 export const CLOSE_STORY_MODAL = 'CLOSE_STORY_MODAL';
 export const SET_SEARCH_TERM = 'SET_SEARCH_TERM';
 export const LOAD_SEARCH_RESULTS = 'LOAD_SEARCH_RESULTS';
+export const SET_CONTEXTS = 'SET_CONTEXTS';
+export const SET_CONTEXT_IS_USER_SELECTED = 'SET_CONTEXT_IS_USER_SELECTED';
+
+export function selectInitialContextById(contextId) {
+  return (dispatch, getState) => {
+    const selectedContext = getContextById(getState(), contextId);
+
+    dispatch({
+      type: LOAD_INITIAL_CONTEXT,
+      payload: selectedContext
+    });
+
+    document.title = getPageTitle(getState());
+  };
+}
+
+export function setContextIsUserSelected(contextIsUserSelected) {
+  return {
+    type: SET_CONTEXT_IS_USER_SELECTED,
+    payload: contextIsUserSelected
+  };
+}
+
+export function selectContextById(contextId) {
+  return (dispatch, getState) => {
+    const selectedContext = getContextById(getState(), contextId);
+
+    dispatch({
+      type: SET_CONTEXT,
+      payload: selectedContext
+    });
+
+    dispatch(setContextIsUserSelected(true));
+
+    dispatch({
+      type: SET_CONTEXT_IS_USER_SELECTED,
+      payload: true
+    });
+
+    if (getState().location.type === 'tool') {
+      dispatch(loadToolDataForCurrentContext());
+    }
+
+    document.title = getPageTitle(getState());
```

Comment:
I wonder if that could be easily done somewhere in the router.

Reply:
Don't think so. The router only calls this when the page changes, which happens prior to this action.